src/dumpling-service/dumpling.web/App_Start/AuthConfig.cs

```diff
@@ -0,0 +1,83 @@
+﻿using DumplingLib;
+using Microsoft.Azure.KeyVault;
+using Microsoft.IdentityModel.Clients.ActiveDirectory;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Cryptography.X509Certificates;
+using System.Threading.Tasks;
+using System.Web;
+using System.Web.Configuration;
+
+namespace dumplingWeb.App_Start
+{
+    public class AuthConfig
+    {
+
+        public static class CertificateHelper
+        {
+            public static X509Certificate2 FindCertificateByThumbprint(string findValue)
+            {
+                X509Store store = new X509Store(StoreName.My, StoreLocation.LocalMachine);
+                try
+                {
+                    store.Open(OpenFlags.ReadOnly);
+
+                    X509Certificate2Collection col = store.Certificates.Find(X509FindType.FindByThumbprint, findValue, false); // TODO CHANGE THIS FACT: Don't validate certs, since the test root isn't installed.
+
+                    if (col == null || col.Count == 0)
+                        return null;
+                    return col[0];
+                }
+                finally
+                {
+                    store.Close();
+                }
+            }
+        }
+
+        private static AuthenticationContext _context;
+        
+        public static async Task<string> GetAccessToken(string authority, string resource, string scope)
+        {
+            _context = new AuthenticationContext(authority, TokenCache.DefaultShared);
+            
+
+            var result = await _context.AcquireTokenAsync(resource, AssertionCert);
+
+            return result.AccessToken;
+        }
+
+        private static ClientAssertionCertificate _cert;
+        public static ClientAssertionCertificate AssertionCert
+        {
+            get
+            {
+                if (_cert == null)
+                {
+                    var clientAssertionCertPfx = CertificateHelper.FindCertificateByThumbprint(WebConfigurationManager.AppSettings["thumbprint"]);
+                    AssertionCert = new ClientAssertionCertificate(WebConfigurationManager.AppSettings["dumpling_ad_client_id"], clientAssertionCertPfx);
```

Comment:
... (see comment above)  here you do not.  if you always use _cert, or always use AssertionCert, then code will be more readable.  Using AssertionCert should allow you to get rid of _cert altogether I believe.